CefSharp.Core/Internals/ClientAdapter.cpp

```diff
@@ -323,15 +350,36 @@ namespace CefSharp
 
         bool ClientAdapter::OnBeforeBrowse(CefRefPtr<CefBrowser> browser, CefRefPtr<CefFrame> frame, CefRefPtr<CefRequest> request, bool isRedirect)
         {
-            IRequestHandler^ handler = _browserControl->RequestHandler;
-            if (handler == nullptr)
+            if (browser->IsPopup())
             {
-                return false;
+                auto popupHandler = _browserControl->PopupHandler;
+                if (popupHandler != nullptr)
+                {
+                    auto browserId = browser->GetIdentifier();
+                    IBrowser^ entry;
+                    if (_popupBrowsers->TryGetValue(browserId, entry))
+                    {
+                        auto frameWrapper = gcnew CefFrameWrapper(frame, _browserAdapter);
+                        CefRequestWrapper^ wrapper = gcnew CefRequestWrapper(request);
+                        return popupHandler->OnBeforeBrowse(_browserControl, entry, wrapper, isRedirect, gcnew CefFrameWrapper(frame, _browserAdapter));
```

Comment:
What's the ownership contract here do you think? Should we own disposing this instance right after the callbacks return, or should we just cache them all thus allowing the handlers the least amount of pain and suffering? Thoughts, @amaitland ?

(I'm talking about the `frameWrapper` here that gets created in this chunk of code.)